pydrive2/files.py

```diff
@@ -687,6 +685,24 @@ def _WrapRequest(self, request):
         """
         if self.http:
             request.http = self.http
+        request = self._AddResourceKeyHeaders(request)
+        return request
+
+    def _AddResourceKeyHeaders(self, request):
+        """Add resourceKey headers to request if file is secured with resourceKey and
+        its available (from a list for example).
+
+        :param request: request to add headers to.
+        :type request: googleapiclient.http.HttpRequest
+        """
+        file_id = self.metadata.get("id") or self.get("id")
+        resourceKey = self.metadata.get("resourceKey") or self.get(
```

Comment:
I assume that `self.metadata.get("resourceKey")` is not needed (and it might trigger an extra call since metadata is "lazy"). key should be provided explicitly before, in the CreateFile, right?